liquibase-couchbase/src/test/java/liquibase/ext/couchbase/change/RemoveDocumentsChangeTest.java

```diff
@@ -111,11 +112,35 @@ void Should_generate_statement_correctly() {
 
     }
 
+    @Test
+    void Should_generate_statement_with_where_clause_correctly() {
+        RemoveDocumentsChange change = createRemoveDocumentChangeWithWhereClause();
+
+        SqlStatement[] statements = change.generateStatements();
+
+        assertThat(statements).hasSize(1);
+        assertThat(statements[0]).isInstanceOf(RemoveDocumentsQueryStatement.class);
+
+        RemoveDocumentsQueryStatement actualStatement = (RemoveDocumentsQueryStatement) statements[0];
+        assertThat(actualStatement.getKeyspace()).isEqualTo(
+                keyspace(change.getBucketName(), change.getScopeName(), change.getCollectionName()));
+        assertThat(actualStatement.getIds()).isEqualTo(change.getIds());
+        assertThat(actualStatement.getWhereCondition()).isEqualTo(change.getWhereCondition());
+    }
+
     private RemoveDocumentsChange createRemoveDocumentChange() {
         return RemoveDocumentsChange.builder().bucketName(TEST_BUCKET)
                 .scopeName(TEST_SCOPE).collectionName(TEST_COLLECTION)
                 .ids(Sets.newHashSet(new Id("id1"), new Id("id2"))).build();
     }
 
+    private RemoveDocumentsChange createRemoveDocumentChangeWithWhereClause() {
+        return RemoveDocumentsChange.builder().bucketName(TEST_BUCKET)
+                .scopeName(TEST_SCOPE).collectionName(TEST_COLLECTION)
+                .ids(Sets.newHashSet(new Id("id1"), new Id("id2")))
```

Comment:
Ids set could be extracted to constant because they're used at least twice

Reply:
done